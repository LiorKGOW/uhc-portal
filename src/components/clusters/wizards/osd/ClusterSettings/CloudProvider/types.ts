import { WifConfig } from '~/components/clusters/wizards/osd/ClusterSettings/CloudProvider/tempWifTypes/WifConfig';

export enum CloudProviderType {
  Aws = 'aws',
  Gcp = 'gcp',
}

export enum GCPAuthType {
  WorkloadIdentityFederation = 'workloadIdentityFederation',
  ServiceAccounts = 'serviceAccounts',
}

export interface WifConfigList {
  items: WifConfig[];
  page: number;
  size: number;
  total: number;
}
