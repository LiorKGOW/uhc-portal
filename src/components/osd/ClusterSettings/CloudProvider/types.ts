export enum CloudProviderType {
  Aws = 'aws',
  Gcp = 'gcp',
}
