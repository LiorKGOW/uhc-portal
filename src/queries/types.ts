import React from 'react';

import { Subscription } from '~/types/accounts_mgmt.v1';
import { ErrorDetail } from '~/types/types';

export type SubscriptionResponseType = {
  subscription: Subscription;
  isAROCluster: boolean;
  isROSACluster: boolean;
  isOSDCluster: boolean;
};

export type RQApiErrorType = {
  pending?: boolean;
  fulfilled?: false;
  error?: true;
  errorCode?: number;
  internalErrorCode?: string;
  operationID?: string;
  errorMessage?: string | React.ReactElement;
  errorDetails?: ErrorDetail[];
};
