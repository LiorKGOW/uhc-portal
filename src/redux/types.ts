import { RouterState } from 'connected-react-router';

export interface ApiRequest {
  details: string;
  error: boolean;
  errorDetails: string;
  errorMessage: string;
  fulfilled: boolean;
  pending: boolean;
}

export interface QuotaList {
  items: any[];
}

export interface Organization extends ApiRequest {
  quotaList: QuotaList;
}

export interface UserProfile {
  organization: Organization;
}

export interface Modal {
  data: any;
}

export interface RosaApi {
  getAWSAccountIDsResponse: {
    data: any[];
  };
}

export interface CcsCredentialsValidity extends ApiRequest {
  cloudProvider: string;
  credentials: any;
}

export interface CcsInquiries {
  ccsCredentialsValidity: CcsCredentialsValidity;
}

export interface GlobalState {
  modal: Modal;
  rosaReducer: RosaApi;
  features: Record<string, boolean>;
  userProfile: UserProfile;
  router: RouterState;
  ccsInquiries: CcsInquiries;
}
