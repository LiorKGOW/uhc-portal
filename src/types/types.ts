import type React from 'react';
import type { FeaturesSupportsLevel } from '@openshift-assisted/ui-lib/ocm';
import type { Cluster as AICluster } from '@openshift-assisted/types/assisted-installer-service';
import type { List, OneMetric, Subscription } from './accounts_mgmt.v1';
import type {
  AWS,
  Cluster,
  ClusterState,
  ClusterStatus,
  LimitedSupportReason,
  Subnetwork,
  VersionGateAgreement,
} from './clusters_mgmt.v1';

export type ViewOptions = {
  currentPage: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
  filter:
    | string
    | { description?: string; loggedBy?: string; timestampFrom?: string; timestampTo?: string };
  sorting: {
    sortField: string;
    isAscending: boolean;
    sortIndex?: number;
  };
  flags: {
    [flag: string]: any;
  };
};

// picking specific Cluster properties to satisfy requirements for
export type FakeCluster = // AICluster &
  Pick<
    Cluster,
    | 'id'
    | 'cloud_provider'
    | 'region'
    | 'console'
    | 'creation_timestamp'
    | 'openshift_version'
    | 'managed'
    | 'ccs'
    | 'external_id'
    | 'inflight_checks'
    | 'name'
    | 'version'
  > & {
    metrics: OneMetric;
    state?: string | ClusterState;
    ['subscription_id']?: string;
    ['activity_timestamp']?: string;
    ['cpu_architecture']?: string;
    product?: Cluster['product'] & { type?: string };
  };

export type ClusterFromSubscription = FakeCluster & {
  subscription?: Subscription;
};

export type ClusterWithPermissions = ClusterFromSubscription & {
  canEdit?: boolean;
  canDelete?: boolean;
  partialCS?: boolean;
};
export type SubscriptionWithPermissions = Subscription & {
  canEdit?: boolean;
};

export type SubscriptionWithPermissionsList = List & {
  items?: Array<SubscriptionWithPermissions>;
};

export type AugmentedCluster = ClusterWithPermissions & {
  canEditOCMRoles?: boolean;
  canViewOCMRoles?: boolean;
  canEditClusterAutoscaler?: boolean;
  idpActions?: {
    [action: string]: boolean;
  };
  machinePoolsActions?: {
    [action: string]: boolean;
  };
  upgradeGates?: VersionGateAgreement[];
  aiCluster?: AICluster;
  limitedSupportReasons?: LimitedSupportReason[];
  aiSupportLevels?: FeaturesSupportsLevel;
  status?: ClusterStatus;
};

export type AugmentedClusterResponse = {
  data: AugmentedCluster;
};

export type ErrorDetail = { kind: string; items?: any };

export type ErrorState = {
  pending: boolean;
  fulfilled: false;
  error: true;
  errorCode?: number;
  internalErrorCode?: string;
  errorMessage?: string | React.ReactElement;
  errorDetails?: ErrorDetail[];
  operationID?: string;
};

export type AWSCredentials = Pick<
  AWS,
  'account_id' | 'access_key_id' | 'secret_access_key' | 'sts'
>;

/** A subnet augmented with info from its parent CloudVPC. */
export type AugmentedSubnetwork = Subnetwork & {
  ['vpc_id']: string;
  ['vpc_name']?: string; // presence depends on AWS "Name" tag, not guaranteed.
};

export type ViewSorting = {
  isAscending: boolean;
  sortField: string;
  sortIndex: number;
};

export type ViewFlags = undefined | null | boolean | string[] | { [key: string]: string[] };

export type UserInfo = {
  username: string;
};
