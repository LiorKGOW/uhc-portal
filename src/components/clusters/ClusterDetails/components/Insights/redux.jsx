import { connect } from 'react-redux';
import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
import InsightsTable from './InsightsTable';

const mapDispatchToProps = dispatch => ({
  addNotification: data => dispatch(addNotification(data)),
});

export default connect(null, mapDispatchToProps)(InsightsTable);
