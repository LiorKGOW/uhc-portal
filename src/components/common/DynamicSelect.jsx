import React from 'react';
import PropTypes from 'prop-types';
import {
  FormGroup, FormSelect, FormSelectOption,
} from '@patternfly/react-core';

import ErrorBox from './ErrorBox';

/**
 * Generic select field whose options are fetched dynamically, and might depend on other fields.
 * Concrete behavior to be injected via props.
 */
class DynamicSelect extends React.Component {
  componentDidMount() {
    this.loadIfNeeded();
    if (this.currentValueIrrelevant()) {
      const { input } = this.props;
      input.onChange('');
    }
  }

  componentDidUpdate() {
    this.loadIfNeeded();
    if (this.currentValueIrrelevant()) {
      const { input } = this.props;
      input.onChange('');
    }
  }

  loadIfNeeded = () => {
    const {
      hasDependencies,
      matchesDependencies,
      requestStatus,
      loadData,
    } = this.props;
    if (hasDependencies && !matchesDependencies && !requestStatus.pending) {
      loadData();
    }
  }

  currentValueIrrelevant = () => {
    const {
      hasDependencies, matchesDependencies, requestStatus, items, input,
    } = this.props;
    if (!input.value) {
      // Blank/placeholder always legitimate.
      return false;
    }
    if (!hasDependencies) {
      // Can't make request.
      return true;
    }
    if (matchesDependencies && requestStatus.fulfilled) {
      // Made request and current value is no longer valid.
      return !items.includes(input.value);
    }
    return false;
  }

  render() {
    const {
      requestStatus,
      items,
      matchesDependencies,
      meta,
      input,
      isRequired,
      label,
      labelIcon,
      helperText,
      placeholder,
      emptyPlaceholder,
      noDependenciesPlaceholder,
      requestErrorTitle,
    } = this.props;
    const show = matchesDependencies && requestStatus.fulfilled;

    let options;
    if (show) {
      if (items.length > 0) {
        options = (
          <>
            <FormSelectOption isDisabled isPlaceholder value="" label={placeholder} />
            {items.map(item => (
              <FormSelectOption key={item} value={item} label={item} />
            ))}
          </>
        );
      } else {
        options = (
          <FormSelectOption isDisabled isPlaceholder value="" label={emptyPlaceholder} />
        );
      }
    } else if (requestStatus.pending) {
      options = (
        <FormSelectOption isDisabled value="" label="Loading..." />
      );
    } else {
      options = (
        <FormSelectOption isDisabled value="" label={noDependenciesPlaceholder} />
      );
    }

    // Prevent FormSelect from picking wrong option when valid options changed.
    // https://github.com/patternfly/patternfly-react/issues/5687
    const value = this.currentValueIrrelevant() ? '' : input.value;

    return (
      <FormGroup
        label={label}
        labelIcon={labelIcon}
        validated={meta.touched && meta.invalid ? 'error' : 'default'}
        helperText={helperText}
        helperTextInvalid={meta.error}
        fieldId={input.name}
        isRequired={isRequired}
      >
        {matchesDependencies && requestStatus.error && (
          <ErrorBox message={requestErrorTitle} response={requestStatus} />
        )}
        <FormSelect
          aria-label={label}
          isDisabled={!(show && items.length > 0)}
          {...input}
          value={value}
        >
          {options}
        </FormSelect>
      </FormGroup>
    );
  }
}
DynamicSelect.propTypes = {
  input: PropTypes.object.isRequired,
  // redux-form metadata like error or active states
  meta: PropTypes.object.isRequired,
  isRequired: PropTypes.bool,
  label: PropTypes.string.isRequired,
  labelIcon: PropTypes.node,
  helperText: PropTypes.string,
  placeholder: PropTypes.string.isRequired,
  emptyPlaceholder: PropTypes.string.isRequired,
  noDependenciesPlaceholder: PropTypes.string,
  requestErrorTitle: PropTypes.string.isRequired,
  hasDependencies: PropTypes.bool.isRequired,
  matchesDependencies: PropTypes.bool.isRequired,
  loadData: PropTypes.func.isRequired,
  requestStatus: PropTypes.shape({
    pending: PropTypes.bool.isRequired,
    fulfilled: PropTypes.bool.isRequired,
    error: PropTypes.object,
  }).isRequired,
  items: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default DynamicSelect;
