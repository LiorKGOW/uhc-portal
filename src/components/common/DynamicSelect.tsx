import React from 'react';
import {
  Alert,
  Button,
  FormGroup,
  FormSelect,
  FormSelectOption,
  Text,
  TextContent,
} from '@patternfly/react-core';
import { WrappedFieldMetaProps, WrappedFieldInputProps } from 'redux-form';
import ErrorBox from './ErrorBox';
import { PromiseReducerState } from '~/redux/types';

type Props = {
  input: Pick<WrappedFieldInputProps, 'value' | 'onChange'> & { name?: string };
  meta: Pick<WrappedFieldMetaProps, 'error' | 'touched' | 'invalid'>;
  isRequired?: boolean;
  label: string;
  labelIcon?: React.ReactElement;
  helperText?: string;
  placeholder: string;
  emptyAlertTitle?: React.ReactNode;
  emptyAlertBody?: React.ReactNode;
  refreshButtonText?: React.ReactNode;
  noDependenciesPlaceholder?: string;
  requestErrorTitle: string;
  hasDependencies: boolean;
  matchesDependencies: boolean;
  loadData: () => void;
  requestStatus: PromiseReducerState;
  items: string[];
};

/**
 * Generic select field whose options are fetched dynamically, and might depend on other fields.
 * Concrete behavior to be injected via props.
 */
class DynamicSelect extends React.Component<Props> {
  componentDidMount() {
    this.loadIfNeeded();
    if (this.currentValueIrrelevant()) {
      const { input } = this.props;
      input.onChange('');
    }
  }

  componentDidUpdate() {
    this.loadIfNeeded();
    if (this.currentValueIrrelevant()) {
      const { input } = this.props;
      input.onChange('');
    }
  }

  loadIfNeeded = () => {
    const { hasDependencies, matchesDependencies, requestStatus, loadData } = this.props;
    if (hasDependencies && !matchesDependencies && !requestStatus.pending) {
      loadData();
    }
  };

  currentValueIrrelevant = () => {
    const { hasDependencies, matchesDependencies, requestStatus, items, input } = this.props;
    if (!input.value) {
      // Blank/placeholder always legitimate.
      return false;
    }
    if (!hasDependencies) {
      // Can't make request.
      return true;
    }
    if (matchesDependencies && requestStatus.fulfilled) {
      // Made request and current value is no longer valid.
      return !items.includes(input.value);
    }
    return false;
  };

  render() {
    const {
      loadData,
      requestStatus,
      items,
      matchesDependencies,
      meta,
      input,
      isRequired,
      label,
      labelIcon,
      helperText,
      placeholder,
      emptyAlertTitle,
      emptyAlertBody,
      refreshButtonText,
      noDependenciesPlaceholder,
      requestErrorTitle,
    } = this.props;
    const show = matchesDependencies && requestStatus.fulfilled;

    let error = null;
    let options = null;

    if (show) {
      if (items.length > 0) {
        options = (
          <>
            <FormSelectOption isDisabled isPlaceholder value="" label={placeholder} />
            {items.map((item) => (
              <FormSelectOption key={item} value={item} label={item} />
            ))}
          </>
        );
      } else {
        error = (
          <Alert isInline variant="danger" title={emptyAlertTitle}>
            <TextContent>
              {emptyAlertBody}
              {refreshButtonText && (
                <Text>
                  <Button variant="secondary" onClick={loadData}>
                    {refreshButtonText}
                  </Button>
                </Text>
              )}
            </TextContent>
          </Alert>
        );
        options = <FormSelectOption isDisabled isPlaceholder value="" label="" />;
      }
    } else if (requestStatus.pending) {
      options = <FormSelectOption isDisabled value="" label="Loading..." />;
    } else if (matchesDependencies && requestStatus.error) {
      error = <ErrorBox message={requestErrorTitle} response={requestStatus} />;
      options = <FormSelectOption isDisabled isPlaceholder value="" label="" />;
    } else {
      options = <FormSelectOption isDisabled value="" label={noDependenciesPlaceholder || ''} />;
    }

    // Prevent FormSelect from picking wrong option when valid options changed.
    // https://github.com/patternfly/patternfly-react/issues/5687
    const value = this.currentValueIrrelevant() ? '' : input.value;

    return (
      <FormGroup
        label={label}
        labelIcon={labelIcon}
        validated={meta.touched && meta.error ? 'error' : 'default'}
        helperText={helperText}
        helperTextInvalid={meta.error}
        fieldId={input.name}
        isRequired={isRequired}
      >
        {error}
        <FormSelect
          aria-label={label}
          isDisabled={!(show && items.length > 0)}
          validated={meta.touched && meta.error ? 'error' : 'default'}
          {...input}
          value={value}
        >
          {options}
        </FormSelect>
      </FormGroup>
    );
  }
}

export default DynamicSelect;
