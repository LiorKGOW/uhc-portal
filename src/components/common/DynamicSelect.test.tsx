import React from 'react';
import { shallow } from 'enzyme';

import DynamicSelect from './DynamicSelect';
import { PromiseReducerState } from '~/redux/types';

type DynamicSelectProps = React.ComponentProps<typeof DynamicSelect>;

describe('<DynamicSelect>', () => {
  let loadData: DynamicSelectProps['loadData'];
  let onChange: DynamicSelectProps['input']['onChange'];
  let baseProps: DynamicSelectProps;
  beforeEach(() => {
    loadData = jest.fn();
    onChange = jest.fn();
    baseProps = {
      loadData,
      input: {
        onChange,
        value: '',
      },
      meta: {
        invalid: false,
        error: null,
        touched: false,
      },
      label: 'Label',
      placeholder: 'Select foo',
      emptyAlertTitle: 'No foos',
      noDependenciesPlaceholder: 'Must select bar first',
      requestErrorTitle: 'Failed listing foos',
      hasDependencies: true,
      matchesDependencies: true,
      requestStatus: {
        pending: false,
        fulfilled: false,
        error: false,
      },
      items: ['Foo', 'Bar'],
    };
  });

  it('without dependencies', () => {
    const select = shallow(<DynamicSelect {...baseProps} hasDependencies={false} />);
    expect(select).toMatchSnapshot('blank');
    expect(select.find('FormSelectOption').at(0).props().label).toEqual('Must select bar first');
    expect(loadData).not.toBeCalled();
  });

  it('gets data', () => {
    const select = shallow(<DynamicSelect {...baseProps} matchesDependencies={false} />);
    expect(loadData).toBeCalled();

    select.setProps({
      requestStatus: { pending: true, fulfilled: false, error: null },
      matchesDependencies: true,
    });
    expect(select).toMatchSnapshot('loading');
    expect(select.find('FormSelectOption').at(0).props().label).toEqual('Loading...');

    select.setProps({
      requestStatus: { pending: false, fulfilled: true, error: null },
    });
    expect(select).toMatchSnapshot('items');
    expect(select.find('FormSelectOption').at(0).props().label).toEqual('Select foo');
    expect(loadData).toBeCalledTimes(1);
  });

  it('empty data', () => {
    const status: PromiseReducerState = { pending: false, fulfilled: true, error: false };
    const select = shallow(<DynamicSelect {...baseProps} requestStatus={status} items={[]} />);
    expect(loadData).not.toBeCalled();
    expect(select).toMatchSnapshot('empty');
    expect(select.find('FormSelectOption').at(0).props().label).toEqual('');
  });

  it('error, untouched', () => {
    const meta = { invalid: true, error: 'Bad!', touched: false };
    const select = shallow(<DynamicSelect {...baseProps} meta={meta} />);
    expect(select).toMatchSnapshot('no error');
  });

  it('error, touched', () => {
    const meta = { invalid: true, error: 'Bad!', touched: true };
    const select = shallow(<DynamicSelect {...baseProps} meta={meta} />);
    expect(select).toMatchSnapshot('shows error');
  });

  it('resets when selected value no longer valid option', () => {
    const status: PromiseReducerState = { pending: false, fulfilled: true, error: false };
    const select = shallow(
      <DynamicSelect {...baseProps} input={{ value: 'Foo', onChange }} requestStatus={status} />,
    );
    expect(loadData).not.toBeCalled();
    expect(onChange).not.toBeCalled();

    select.setProps({
      items: ['Bar', 'Baz'],
    });
    expect(onChange).toBeCalledWith('');
    select.setProps({
      input: { value: '', onChange },
    });
    expect(select).toMatchSnapshot('new items');
    expect(select.find('FormSelectOption').at(0).props().label).toEqual('Select foo');
  });

  it('resets on lost dependencies', () => {
    const status: PromiseReducerState = { pending: false, fulfilled: true, error: false };
    const select = shallow(
      <DynamicSelect {...baseProps} input={{ value: 'Foo', onChange }} requestStatus={status} />,
    );
    expect(loadData).not.toBeCalled();
    expect(onChange).not.toBeCalled();

    select.setProps({
      hasDependencies: false,
      matchesDependencies: false,
    });
    expect(onChange).toBeCalledWith('');
    select.setProps({
      input: { value: '', onChange },
    });
    expect(select.find('FormSelectOption').at(0).props().label).toEqual('Must select bar first');
  });
});
