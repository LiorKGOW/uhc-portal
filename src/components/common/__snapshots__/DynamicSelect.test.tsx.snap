// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DynamicSelect> empty data: empty 1`] = `
<FormGroup
  helperTextInvalid={null}
  label="Label"
  validated="default"
>
  <Alert
    isInline={true}
    title="No foos"
    variant="danger"
  >
    <TextContent />
  </Alert>
  <FormSelect
    aria-label="Label"
    className=""
    isDisabled={true}
    isIconSprite={false}
    isRequired={false}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    ouiaSafe={true}
    validated="default"
    value=""
  >
    <FormSelectOption
      isDisabled={true}
      isPlaceholder={true}
      label=""
      value=""
    />
  </FormSelect>
</FormGroup>
`;

exports[`<DynamicSelect> error, touched: shows error 1`] = `
<FormGroup
  helperTextInvalid="Bad!"
  label="Label"
  validated="error"
>
  <FormSelect
    aria-label="Label"
    className=""
    isDisabled={true}
    isIconSprite={false}
    isRequired={false}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    ouiaSafe={true}
    validated="error"
    value=""
  >
    <FormSelectOption
      isDisabled={true}
      label="Must select bar first"
      value=""
    />
  </FormSelect>
</FormGroup>
`;

exports[`<DynamicSelect> error, untouched: no error 1`] = `
<FormGroup
  helperTextInvalid="Bad!"
  label="Label"
  validated="default"
>
  <FormSelect
    aria-label="Label"
    className=""
    isDisabled={true}
    isIconSprite={false}
    isRequired={false}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    ouiaSafe={true}
    validated="default"
    value=""
  >
    <FormSelectOption
      isDisabled={true}
      label="Must select bar first"
      value=""
    />
  </FormSelect>
</FormGroup>
`;

exports[`<DynamicSelect> gets data: items 1`] = `
<FormGroup
  helperTextInvalid={null}
  label="Label"
  validated="default"
>
  <FormSelect
    aria-label="Label"
    className=""
    isDisabled={false}
    isIconSprite={false}
    isRequired={false}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    ouiaSafe={true}
    validated="default"
    value=""
  >
    <FormSelectOption
      isDisabled={true}
      isPlaceholder={true}
      label="Select foo"
      value=""
    />
    <FormSelectOption
      key="Foo"
      label="Foo"
      value="Foo"
    />
    <FormSelectOption
      key="Bar"
      label="Bar"
      value="Bar"
    />
  </FormSelect>
</FormGroup>
`;

exports[`<DynamicSelect> gets data: loading 1`] = `
<FormGroup
  helperTextInvalid={null}
  label="Label"
  validated="default"
>
  <FormSelect
    aria-label="Label"
    className=""
    isDisabled={true}
    isIconSprite={false}
    isRequired={false}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    ouiaSafe={true}
    validated="default"
    value=""
  >
    <FormSelectOption
      isDisabled={true}
      label="Loading..."
      value=""
    />
  </FormSelect>
</FormGroup>
`;

exports[`<DynamicSelect> resets when selected value no longer valid option: new items 1`] = `
<FormGroup
  helperTextInvalid={null}
  label="Label"
  validated="default"
>
  <FormSelect
    aria-label="Label"
    className=""
    isDisabled={false}
    isIconSprite={false}
    isRequired={false}
    onBlur={[Function]}
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            "",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onFocus={[Function]}
    ouiaSafe={true}
    validated="default"
    value=""
  >
    <FormSelectOption
      isDisabled={true}
      isPlaceholder={true}
      label="Select foo"
      value=""
    />
    <FormSelectOption
      key="Bar"
      label="Bar"
      value="Bar"
    />
    <FormSelectOption
      key="Baz"
      label="Baz"
      value="Baz"
    />
  </FormSelect>
</FormGroup>
`;

exports[`<DynamicSelect> without dependencies: blank 1`] = `
<FormGroup
  helperTextInvalid={null}
  label="Label"
  validated="default"
>
  <FormSelect
    aria-label="Label"
    className=""
    isDisabled={true}
    isIconSprite={false}
    isRequired={false}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    ouiaSafe={true}
    validated="default"
    value=""
  >
    <FormSelectOption
      isDisabled={true}
      label="Must select bar first"
      value=""
    />
  </FormSelect>
</FormGroup>
`;
